File: cs-499 project java/Contact/ContactService.java
/*
 * Name: Justin Fifarek
 * Course: SNHU CS-320
 * Date: 11/16/2023
 * 
 * Module 3-2 Milestone - Contact Service class
 */

package Contact;

import java.util.ArrayList;

public class ContactService {
	// Array list to hold contacts
	public ArrayList<Contact> contactList = new ArrayList<Contact>();
	
	// Display list of stored contacts
	public void displayContacts() {
		for (int i = 0; i < contactList.size(); i++) {
			System.out.println(" Contact ID: " + contactList.get(i).getContactID());
			System.out.println(" First Name: " + contactList.get(i).getFirstName());
			System.out.println(" Last Name: " + contactList.get(i).getLastName());
			System.out.println(" Phone Number: " + contactList.get(i).getPhone());
			System.out.println(" Address: " + contactList.get(i).getAddress());
			for (int x = 0; x < contactList.get(i).getAddress().length() + 9; x++) {
				System.out.print("-");
			}
			System.out.println("");
		}
	}
	
	// Adds new contact using contact constructor
	public void addContact(String firstName, String lastName, String phone, String address) {
		Contact contact = new Contact(firstName, lastName, phone, address);
		contactList.add(contact);
	}
	
	// Searches through the array and returns contact information, given a contact ID
	// Returns a default contact if no matching ID is found
	public Contact getContact(String contactID) {
		Contact contact = new Contact(null, null, null, null);
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().contentEquals(contactID)) {
				contact = contactList.get(i);
			}
		}
		return contact;
	}
	
	// Searches contact array given a contactID and deletes the entry
	public void deleteContact(String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.remove(i);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
	
	// Updates contact firstName of given contactID with given string
	public void updateFirstName(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setFirstName(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
	
	// Updates contact lastName of given contactID with given string
	public void updateLastName(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setLastName(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
	
	// Updates contact phone number of given contactID with given string
	public void updatePhone(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setPhone(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
	
	// Updates contact address of given contactID with given string
	public void updateAddress(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setAddress(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
}

File: cs-499 project c++/Contact/ContactService.cpp
/*
    Name: Justin Fifarek
    Project: ContactService class
*/
#include <atomic>
#include <vector>
#include "Contact.h"
#include "ContactService.h"

// Display list of stored contacts
void ContactService::displayContacts() {
    for (size_t i = 0; i < contactList.size(); ++i) {
        std::cout << " Contact ID: " << contactList[i].getContactID() << std::endl;
        std::cout << " First Name: " << contactList[i].getFirstName() << std::endl;
        std::cout << " Last Name: " << contactList[i].getLastName() << std::endl;
        std::cout << " Phone Number: " << contactList[i].getPhone() << std::endl;
        std::cout << " Address: " << contactList[i].getAddress() << std::endl;
        for (size_t x = 0; x < contactList[i].getAddress().length() + 9; ++x) {
            std::cout << "-";
        }
        std::cout << std::endl;
    }
}

// Adds new contact using the contact constructor
void ContactService::addContact(std::string firstName, std::string lastName, std::string phone, std::string address) {
    Contact contact(firstName, lastName, phone, address);
    contactList.push_back(contact);
}

// Searches through contactList vector and returns the contact
// of given contactID, if it exists.
Contact ContactService::getContact(std::string contactID) {
    for (size_t i = 0; i < contactList.size(); ++i) {
        if (contactList[i].getContactID() == contactID) {
            return contactList[i];
        }
    }
    return Contact("", "", "", "");
}

// Searches through contactList vector and deletes the contact
// of given contactID, if it exists.
void ContactService::deleteContact(std::string contactID) {
    for (auto it = contactList.begin(); it != contactList.end(); ++it) {
        if (it->getContactID() == contactID) {
            contactList.erase(it);
            return;
        }
    }
}

// Searches through contactList vector and updates the firstName
// of given contactID, if it exists.
void ContactService::updateFirstName(std::string updatedString, std::string contactID) {
    for (size_t i = 0; i < contactList.size(); ++i) {
        if (contactList[i].getContactID() == contactID) {
            contactList[i].setFirstName(updatedString);
            return;
        }
    }
    std::cout << "Contact ID: " << contactID << " not found." << std::endl;
}

// Searches through contactList vector and updates the lastName
// of given contactID, if it exists.
void ContactService::updateLastName(std::string updatedString, std::string contactID) {
    for (size_t i = 0; i < contactList.size(); ++i) {
        if (contactList[i].getContactID() == contactID) {
            contactList[i].setLastName(updatedString);
            return;
        }
    }
    std::cout << "Contact ID: " << contactID << " not found." << std::endl;
}

// Searches through contactList vector and updates the phone
// of given contactID, if it exists.
void ContactService::updatePhone(std::string updatedString, std::string contactID) {
    for (size_t i = 0; i < contactList.size(); ++i) {
        if (contactList[i].getContactID() == contactID) {
            contactList[i].setPhone(updatedString);
            return;
        }
    }
    std::cout << "Contact ID: " << contactID << " not found." << std::endl;
}

// Searches through contactList vector and updates the address
// of given contactID, if it exists.
void ContactService::updateAddress(std::string updatedString, std::string contactID) {
    for (size_t i = 0; i < contactList.size(); ++i) {
        if (contactList[i].getContactID() == contactID) {
            contactList[i].setAddress(updatedString);
            return;
        }
    }
    std::cout << "Contact ID: " << contactID << " not found." << std::endl;
};